extensions/attention-stream/src/background.ts

```diff
@@ -154,12 +157,60 @@ async function stateChangeCallback(newState) {
 
       webScience.pageNavigation.onPageData.addListener(this.pageDataListener, { matchPatterns: ["<all_urls>"] });
 
+      const matchPatterns = webScience.matching.domainsToMatchPatterns(destinationDomains, true);
+
+      // Handle article content callbacks.
+      this.pageTextListener = async (pageData) => {
+        articleContents.pageId.set(pageData.pageId);
+        articleContents.url.setUrl(pageData.url);
+        articleContents.title.set(pageData.title);
+        articleContents.textContent.set(pageData.textContent);
+
+        attentionStreamPings.articleContents.submit();
+      };
+
+      webScience.pageText.onTextParsed.addListener(this.pageTextListener, { matchPatterns });
+
+      // Register the content script for measuring advertisement info.
+      // The CSS selectors file is needed to find ads on the page.
+      // Load content script(s) required by this extension.
+      await browser.scripting.registerContentScripts([{
+        id: "page-ads",
+        js: ["dist/browser-polyfill.min.js", "dist/page-ads.content.js"],
+        matches: matchPatterns,
+        persistAcrossSessions: false,
+        runAt: "document_idle"
+      }]);
+
+      this.advertisementListener = async (adInfo, sender) => {
+        advertisements.pageId.set(adInfo.pageId);
+        advertisements.url.setUrl(webScience.matching.normalizeUrl(sender.url));
+        advertisements.body.set(JSON.stringify(adInfo.body));
+        advertisements.ads.set(JSON.stringify(adInfo.ads));
+
+        attentionStreamPings.advertisements.submit();
+      }
+
+      // Handle advertisement callbacks.
+      webScience.messaging.onMessage.addListener(this.advertisementListener, {
+        type: "WebScience.advertisements",
```

Comment:
Also, I can't figure out where this message is being sent from!

Reply:
It's sent by the `page-ads.content.ts` which apparently I didn't actually commit in this branch, sorry! It's derived from the content script beyond-the-paywall uses to detect ads, it's essentially exactly the same just some minor typescript conversion related changes.

Reply:
I just pushed this thanks for pointing that out!